File: 백준/Bronze/25304. 영수증/영수증.py
import sys

input = sys.stdin.readline
amount = int(input())
n = int(input())
sum = 0
for i in range(n):
    price, ea = map(int, input().split())
    sum = sum + (price * ea)

if amount == sum:
    print("Yes")
else:
    print("No")

File: 백준/Unrated/25304. 영수증/영수증.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.sql.SQLOutput;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        int amount = Integer.parseInt(br.readLine());
        int count = Integer.parseInt(br.readLine());
        int sum = 0;

        for (int i = 0; i < count; i++) {
            st = new StringTokenizer(br.readLine());
            sum += Integer.parseInt(st.nextToken()) * Integer.parseInt(st.nextToken());
        }
        if (amount == sum) System.out.println("Yes");
        else System.out.println("No");
    }
}